vpr/src/base/read_route.cpp

```diff
@@ -281,8 +282,9 @@ static void process_nodes(std::ifstream& fp, ClusterNetId inet, const char* file
                 }
             }
 
-            ptc = atoi(tokens[5 + offset].c_str());
-            if (device_ctx.rr_graph.node_ptc_num(node) != ptc) {
+            int ptc0, ptc1, ptc2, ptc3;
+            format_ptc_num(ptc0, ptc1, ptc2, ptc3, tokens[5 + offset], inet, filename, lineno);
+            if (device_ctx.rr_graph.node_ptc_num(node) != ptc0) {
```

Comment:
Need a double check on the selection of ptc number. I forgot if we should always use the first one or the last one. For nodes whose direction is INC_DIR, I think we should use the first one. However, for the nodes whose direction is DEC_DIR, please double check if this is true or not.